tests/pytests/unit/utils/test_pycrypto.py

```diff
@@ -0,0 +1,190 @@
+import contextlib
+import logging
+import re
+
+import pytest
+import salt.utils.platform
+import salt.utils.pycrypto
+from salt.exceptions import SaltInvocationError
+from tests.support.mock import patch
+
+passwd = "test_password"
+invalid_salt = "thissaltistoolong" * 10
+expecteds = {
+    "sha512": {
+        "hashed": "$6$rounds=65601$goodsalt$lZFhiN5M8RTLd9WKDin50H4lF4F8HGMIdwvKs.nTG7f8F0Y4P447Zb9/E8SkUWjY.K10QT3NuHZNDgc/P/NjT1",
+        "salt": "rounds=65601$goodsalt",
+        "badsalt": "badsalt",
+    },
+    "sha256": {
+        "hashed": "$5$rounds=53501$goodsalt$W.uoco0wMfGLDOlsbW52E6raFS1Nhj0McfUTj2vORt7",
+        "salt": "rounds=53501$goodsalt",
+        "badsalt": "badsalt",
+    },
+    "blowfish": {
+        "hashed": "$2b$10$goodsaltgoodsaltgoodsObFfGrJwfV.13QddrZIh2w1ccESmvj8K",
+        "salt": "10$goodsaltgoodsaltgoodsa",
+        "badsalt": "badsaltbadsaltbadsaltb",
+    },
+    "md5": {
+        "hashed": "$1$goodsalt$4XQMx4a4e1MpBB8xzz.TQ0",
+        "salt": "goodsalt",
+        "badsalt": "badsalt",
+    },
+    "crypt": {"hashed": "goVHulDpuGA7w", "salt": "go", "badsalt": "ba"},
+}
+
+
+@pytest.fixture(params=["sha512", "sha256", "blowfish", "md5", "crypt"])
+def algorithm(request):
+    return request.param
+
+
+@pytest.mark.skipif(not salt.utils.pycrypto.HAS_CRYPT, reason="crypt not available")
+@patch("salt.utils.pycrypto.methods", {})
```

Comment:
Let's use `with patch()` in the function body instead please 

https://engblog.nextdoor.com/what-your-mocks-do-when-you-aren-t-looking-b278e0d9e201